api/organisations/permissions/permissions.py

```diff
@@ -88,6 +88,17 @@ class OrganisationPermission(BasePermission):
     def has_permission(self, request, view):
         if view.action == "create" and settings.RESTRICT_ORG_CREATE_TO_SUPERUSERS:
             return request.user.is_superuser
+
+        organisation_id = view.kwargs.get("pk")
+        if organisation_id and not organisation_id.isnumeric():
+            raise ValidationError("Invalid organisation ID")
+
+        if view.action == "remove_users":
+            return request.user.is_organisation_admin(int(organisation_id))
```

Comment:
Sorry - this will crash the view if `organisation_id` is `None`.

Reply:
Which it can't be if `view.action == "remove_users"` - it's a detail route.

Reply:
What's the point of `if organisation_id` on L93 then?

Reply:
See https://github.com/Flagsmith/flagsmith/pull/3791/files#r1569344842.

Reply:
Because `has_permission` also covers non-retail routes (e.g. create, list, etc.)